lib/fastlane/plugin/wpmreleasetoolkit/actions/ios/ios_generate_strings_file_from_code.rb

```diff
@@ -2,15 +2,32 @@ module Fastlane
   module Actions
     class IosGenerateStringsFileFromCodeAction < Action
       def self.run(params)
-        files = files_matching(paths: params[:paths], exclude: params[:exclude])
-        flags = [('-q' if params[:quiet]), ('-SwiftUI' if params[:swiftui])].compact
-        flags += Array(params[:routines]).flat_map { |routine| ['-s', routine] }
-        cmd = ['genstrings', '-o', params[:output_dir], *flags, *files]
-        out = Actions.sh_control_output(*cmd, print_command: FastlaneCore::Globals.verbose?, print_command_output: true)
-        out = out.scrub.strip.split("\n")
-        errors = out.select { |line| line.include?('genstrings: error: ') }
-        UI.user_error!(errors.join("\n")) unless !params[:fail_on_error] || errors.empty?
-        out
+        output_encoding = begin
+          Encoding.find(params[:output_encoding])
+        rescue ArgumentError => e
+          UI.user_error!(e.message)
```

Comment:
Maybe we could use here a custom message instead, as the reason for failure is clear to us?
```suggestion
          UI.user_error!("Invalid output encoding '#{params[:output_encoding]}'. Please provide a valid encoding supported by Ruby.")
```

Reply:
I thought about it, but also thought that given `rescue ArgumentError` is quite generic,  there's technically no guarantee that the only `ArgumentError` that `Encoding.find` could possibly `raise` will _always_ be about "invalid encoding - {name}" and never about any other `ArgumentError` message…

Reply:
Yeah, also works for me.
I was thinking pretty much anything will be about the argument and "invalid encoding" has a generic meaning here, not necessarily "we don't know this encoding". But I'm indifferent in this case as I also expect any error should be clear enough 👍